src/main/java/de/nplay/moderationbot/config/bot/BotConfig.java

```diff
@@ -0,0 +1,67 @@
+package de.nplay.moderationbot.config.bot;
+
+import de.nplay.moderationbot.config.ConfigService;
+import de.nplay.moderationbot.config.bot.type.BotConfigType;
+import net.dv8tion.jda.api.JDA;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class BotConfig {
+
+    private final String key;
+    private final String displayName;
+    private final BotConfigType type;
+
+    BotConfig(String key, String displayName, BotConfigType type) {
+        this.key = key;
+        this.displayName = displayName;
+        this.type = type;
+    }
+
+    public String key() {
+        return key;
+    }
+
+    public String displayName() {
+        return displayName;
+    }
+
+    public Optional<String> value() {
+        return ConfigService.get(key);
+    }
+
+    public Optional<String> formattedValue() {
+        return value().map(type.formatter());
+    }
+
+    public BotConfigType type() {
+        return type;
+    }
+
+    public Boolean validate(String value) {
+        return type.validator().apply(value);
+    }
+
+    public static Collection<BotConfig> getConfigs(JDA jda) {
+        return Arrays.stream(BotConfigs.class.getMethods())
+                .filter(method -> method.getReturnType().equals(BotConfig.class))
```

Comment:
can be removed if using `getDeclaredMethods` and is also inconsistent because we dont check the method signature, accessiblity, etc